### DISCUSSION TITLE
Using BlazorHeroUser as a related entity

### DISCUSSION THREAD

**Discussion by dMiracleMan** [fullstackhero/blazor-starter-kit#339] [Q&A]
Is there a way to include a relationship between entities in the domain model to the BlazorHeroUser?  I tried adding it but it wont find the reference.

**Comment by neozhu** [comment]
BlazorHeroContext inherits from IdentityDbContex
so, you can use BlazorHeroContext.Users or _db.Users implementing associated query with linq
![image](https://user-images.githubusercontent.com/1549611/137048249-e1c37332-42da-4f8f-9bd3-7f84edf082ae.png)

**Reply by dMiracleMan** [reply]
am i able to use it a the application level? in the features?

**Reply by neozhu** [reply]
yes, you can  write it  `_unitOfWork.Repository<Users>().GetByIdAsync(id);`  or direct dependency injection BlazorHeroContext

**Reply by dMiracleMan** [reply]
In the application project I dont have the Infrastructure project reference to add the BlazorHeroContext, do I need to add it?

In this sample, _unitOfWork.Repository<Users>().GetByIdAsync(id);  --> should it be BlazorHeroUser or Users? I cant objects from the identity, it wont recognize them.

**Reply by ajakblackgoat** [reply]
Adding a navigation prop in domain entity pointing to BlazorHeroUser definitely cannot be done because domain proj can't see Infra. Adding Infra project as reference also not possible because it will result in a reference loop (Infra already have reference to domain proj).

I ended up having to use ```_userManager.GetAsync(_data.CreatedBy)``` in razor page which is to display a single entity. However, for display in a MudTable, must find other way.

**Reply by sergeytys** [reply]
Does anyone know what was the logic to put the BlazorHeroUser in the Infra proj anyway? I can't figure out why ... maybe put it in the Domain (all projs should refer to it in the end anyway...) - would't that solve that issue? (if one refactors this in its own fork)?  or there are argements againts such a move I'm missing?

**Reply by biproberkay** [reply]
i guess, it is a architectural concern. having a package reference doesnt recommended for Doman layer. and the BlazorUser.cs inherits from Identityuser. for doing this domain must include the required identity package.
